### PR TITLE
Update Pyomo.DoE documention

### PR DESCRIPTION
**Pull request by adowling2**
## Summary/Motivation:
- Minor updates to Pyomo.DoE documentation

For convenience, here is the most recent documentation: https://pyomo.readthedocs.io/en/latest/contributed_packages/doe/doe.html

## Changes proposed in this PR:
- Streamlined documentation, fixed a few typos
- Added Pyomo.DoE to the documentation page
- Fixed sphinx warnings

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by adowling2** [comment]
Here are the warning messages I get when compiling the Pyomo.DoE documentation locally:
```
/Users/adowling/DowlingLab/pyomo/pyomo/contrib/doe/doe.py:docstring of pyomo.contrib.doe.doe.DesignOfExperiments.compute_FIM:3: ERROR: Unexpected indentation.
/Users/adowling/DowlingLab/pyomo/pyomo/contrib/doe/doe.py:docstring of pyomo.contrib.doe.doe.DesignOfExperiments.run_grid_search:4: ERROR: Unexpected indentation.
/Users/adowling/DowlingLab/pyomo/pyomo/contrib/doe/measurements.py:docstring of pyomo.contrib.doe.measurements.Measurements.__init__:13: WARNING: Definition list ends without a blank line; unexpected unindent.
WARNING: autodoc: failed to import class 'FIM_result' from module 'pyomo.contrib.doe.result'; the following exception was raised:
Traceback (most recent call last):
  File "/Users/adowling/anaconda3/envs/pyomodev/lib/python3.10/site-packages/sphinx/util/inspect.py", line 380, in safe_getattr
    return getattr(obj, name, *defargs)
AttributeError: module 'pyomo.contrib.doe.result' has no attribute 'FIM_result'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/adowling/anaconda3/envs/pyomodev/lib/python3.10/site-packages/sphinx/ext/autodoc/importer.py", line 101, in import_object
    obj = attrgetter(obj, mangled_name)
  File "/Users/adowling/anaconda3/envs/pyomodev/lib/python3.10/site-packages/sphinx/ext/autodoc/__init__.py", line 307, in get_attr
    return autodoc_attrgetter(self.env.app, obj, name, *defargs)
  File "/Users/adowling/anaconda3/envs/pyomodev/lib/python3.10/site-packages/sphinx/ext/autodoc/__init__.py", line 2795, in autodoc_attrgetter
    return safe_getattr(obj, name, *defargs)
  File "/Users/adowling/anaconda3/envs/pyomodev/lib/python3.10/site-packages/sphinx/util/inspect.py", line 396, in safe_getattr
    raise AttributeError(name) from exc
AttributeError: FIM_result

WARNING: autodoc: failed to import class 'Grid_Search_Result' from module 'pyomo.contrib.doe.result'; the following exception was raised:
Traceback (most recent call last):
  File "/Users/adowling/anaconda3/envs/pyomodev/lib/python3.10/site-packages/sphinx/util/inspect.py", line 380, in safe_getattr
    return getattr(obj, name, *defargs)
AttributeError: module 'pyomo.contrib.doe.result' has no attribute 'Grid_Search_Result'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/adowling/anaconda3/envs/pyomodev/lib/python3.10/site-packages/sphinx/ext/autodoc/importer.py", line 101, in import_object
    obj = attrgetter(obj, mangled_name)
  File "/Users/adowling/anaconda3/envs/pyomodev/lib/python3.10/site-packages/sphinx/ext/autodoc/__init__.py", line 307, in get_attr
    return autodoc_attrgetter(self.env.app, obj, name, *defargs)
  File "/Users/adowling/anaconda3/envs/pyomodev/lib/python3.10/site-packages/sphinx/ext/autodoc/__init__.py", line 2795, in autodoc_attrgetter
    return safe_getattr(obj, name, *defargs)
  File "/Users/adowling/anaconda3/envs/pyomodev/lib/python3.10/site-packages/sphinx/util/inspect.py", line 396, in safe_getattr
    raise AttributeError(name) from exc
AttributeError: Grid_Search_Result

/Users/adowling/DowlingLab/pyomo/pyomo/core/base/PyomoModel.py:docstring of pyomo.core.base.block._BlockData.block_data_objects:8: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/adowling/DowlingLab/pyomo/pyomo/core/base/PyomoModel.py:docstring of pyomo.core.base.block._BlockData.block_data_objects:8: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/adowling/DowlingLab/pyomo/pyomo/contrib/appsi/solvers/highs.py:docstring of pyomo.contrib.appsi.solvers.highs.Highs.highs_options:7: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/adowling/DowlingLab/pyomo/pyomo/core/plugins/transform/scaling.py:docstring of pyomo.core.plugins.transform.scaling.ScaleModel:44: ERROR: Unknown directive type "todo".

.. todo:: - implement an option to change the variables names or not
```

@blnicho @jialuw96 @jsiirola suggestions are welcome. I'll continue to investigate.

**Comment by blnicho** [comment]
@adowling2 I think the problem is a couple class names that were changed in the code and never updated in the rst file. I went ahead and fixed them. You can ignore the warnings coming from outside Pyomo.DoE, I'll fix them separately.

**Comment by adowling2** [comment]
@jialuw96 @jsiirola @kaklise This is ready to review. I did some proofreading with the documentation and added the Pyomo.DoE logo. I checked the figure sizing on my local machine. @blnicho Helped fix some doc strings and sphinx warnings.

File: doc/OnlineDocs/contributed_packages/doe/doe.rst [hunk 1]
```diff
@@ -64,14 +71,14 @@ where:
...
 
 .. note::
-    * Process models provided to Pyomo.DoE should define an extra scenario index for all state variables and all parameters, as the first index before any other index.
-    * Process models must include an index for time, named ``t``. For steady-state models, t should be [0].
-    * Measurements can have an extra index besides time.
-    * Parameters and design variables should be defined as Pyomo ``var`` components on the model to use ``direct_kaug`` mode, and can be defined as Pyomo ``Param`` object if not using ``direct_kaug``.
+    * Process models provided to Pyomo.DoE should define an extra scenario index for all state variables and all parameters, as the first index before any other index. The next version of Pyomo.DoE will remove this requirement.
+    * Process models must include an index for time, named ``t``. For steady-state models, ``t`` should be ``[0]``.
+    * Measurements can have an extra index (e.g., spatial domain) besides time.
+    * Parameters and design variables should be defined as Pyomo ``var`` components on the model to use ``direct_kaug`` mode. Other modes allow these to be defines as Pyomo ``Param`` objects.
```
Comment (mrmundt) [inline doc/OnlineDocs/contributed_packages/doe/doe.rst:77]: Small typo: `defined`

File: doc/OnlineDocs/contributed_packages/doe/doe.rst [hunk 2]
```diff
@@ -64,14 +71,14 @@ where:
...
     * Create model function should take scenarios as the first argument of this function.
     * Design variables are defined with and only with a time index.
 
-Based on the above notation, the form of the MBDoE problem addressed in Pyomo.DoE is shown below:
+Pyomo.DoE solves the following DAE-constrainted optimizaiton problem:
```
Comment (mrmundt) [inline doc/OnlineDocs/contributed_packages/doe/doe.rst:81]: Small typo: `optimization`

**Review by mrmundt — CHANGES_REQUESTED** [review]
A couple of typos, but looks good!

File: doc/OnlineDocs/contributed_packages/doe/doe.rst (hunk 1 above)
Comment (adowling2) [reply doc/OnlineDocs/contributed_packages/doe/doe.rst]: Just fixed. Thank you

File: doc/OnlineDocs/contributed_packages/doe/doe.rst (hunk 2 above)
Comment (adowling2) [reply doc/OnlineDocs/contributed_packages/doe/doe.rst]: Just fixed. Thank you.

**Review by mrmundt — APPROVED** [review]

**Comment by DamianAgi** [comment]
Fix typos:
_Location: typo_
- [ ] Fig 2 caption: "is" missing between "Pyomo.DoE" and "focused" 
- [ ] DAE-constrainted optimization problem setup: inconsistent pluralization in definition of design variables. See snapshot 1
- [ ] Pyomo.DoE Required Inputs, line 2: change "value" to "values"
- [ ] Arguments, param_init: inconsistent pluralization, "If **they are an** indexed variable"
- [ ] design_variable_timepoints: inconsistent pluralization, "variable **names** and **its**"
- [ ] sequential_finite: repetition, "Calculates **a one** scenario"
- [ ] Parameters, store_name: spelling of "multiple"
- [ ] Pyomo.DoE Usage Example, Step 4: change "the" to "to"; "Heatmaps (or lines) are used **the** visualize the sensitivity"


Snapshot 1:
typo 1

**Comment by adowling2** [comment]
@DamianAgi Thank you. I fixed all of the typos you found. (Can you mark everything on your list as done? I do not have permission to do that.)

@mrmundt This is ready for a quick re-review.

**Comment by DamianAgi** [comment]
> 

Sure, done!

File: pyomo/contrib/doe/measurements.py [hunk 3]
```diff
@@ -36,20 +36,27 @@
...
         Parameters
         ----------
         measurement_index_time:
             a ``dict``, keys are measurement variable names, 
-                if there are extra index, for e.g., Var[scenario, extra_index, time]:
-                    values are a dictionary, keys are its extra index, values are its measuring time points. 
-                if there are no extra index, for e.g., Var[scenario, time]:
-                    values are a list of measuring time point.
+
+            * if there are extra index, for e.g., Var[scenario, extra_index, time]:
```
[the remark this answers is not in the document]
Comment (adowling2) [reply pyomo/contrib/doe/measurements.py]: Fixed.

File: pyomo/contrib/doe/measurements.py [hunk 4]
```diff
@@ -36,20 +36,27 @@
...
+              values are a dictionary, keys are its extra index, values are its 
+              measuring time points. 
+            * if there are no extra index, for e.g., Var[scenario, time]:
```
[the remark this answers is not in the document]
Comment (adowling2) [reply pyomo/contrib/doe/measurements.py]: Fixed.

**Review by blnicho — APPROVED** [review]

**Review by mrmundt — APPROVED** [review]

**Pull request merged**